packages/report-server/src/reportBuilder/transform/aliases/keyValueByFieldAliases.ts

```diff
@@ -3,37 +3,56 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Row } from '../../types';
+import { TransformTable } from '../table';
+import { FieldValue } from '../../types';
+
+const keyValueByField = (table: TransformTable, field: string) => {
+  const fieldValues = table.getColumnValues(field) as string[];
+  const valueValues = table.getColumnValues('value');
+  const newColumnNames = Array.from(new Set(fieldValues));
+  const newColumns: Record<string, FieldValue[]> = Object.fromEntries(
+    newColumnNames.map(columnName => [
+      columnName,
+      // Flag all values as skip char to avoid overwriting data
+      table.hasColumn(columnName)
```

Comment:
Should `table.hasColumn(columnName)` always be true? `newColumnNames` are built from `table`.